src/library/scala/collection/SeqView.scala

```diff
@@ -30,16 +30,42 @@ trait SeqView[+A] extends SeqOps[A, View, View[A]] with View[A] {
   def concat[B >: A](suffix: SeqView.SomeSeqOps[B]): SeqView[B] = new SeqView.Concat(this, suffix)
   def appendedAll[B >: A](suffix: SeqView.SomeSeqOps[B]): SeqView[B] = new SeqView.Concat(this, suffix)
   def prependedAll[B >: A](prefix: SeqView.SomeSeqOps[B]): SeqView[B] = new SeqView.Concat(prefix, this)
+
+  override def sorted[B >: A](implicit ord: Ordering[B]): SeqView[A] = {
+    val len = this.length // does not evaluate any view operations; checks `underlying.length`
+    new SeqView.Id(
+      if (len == 0) Nil
+      else if (len == 1) List(head)
+      else {
+        val arr = new Array[AnyRef](len)
```

Comment:
Since you can't rely on `knownSize` (and shouldn't call `length`) it would be better to use an `ArrayBuffer`.

Reply:
When relying on `length`, we traverse the collection once without operations to get `length`, and once with operations when copying to the array. When using an `ArrayBuffer`, we traverse with operations when copying to the buffer (which may also involve several array copies), and then traverse without operations when copying to the array. Is that any better?

Potentially the second copy from the collection to the array could use `toArray` or `copyToArray`, which would improve the copy performance for `ArrayBuffer` (and possibly others)?

These changes need to be applied to the base implementation anyway, so I'm not sure whether we should hold this PR up for them. I'm happy to make the changes regardless.